packages/@tinacms/api-git/src/repo.ts

```diff
@@ -0,0 +1,201 @@
+/**
+
+Copyright 2019 Forestry.io Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+import git from 'simple-git/promise'
+
+import * as path from 'path'
+import * as fs from 'fs'
+import { deleteFile, writeFile } from './file-writer'
+import { checkFilePathIsInParent } from './utils'
+
+export interface CommitOptions {
+  files: string[]
+  message: string
+  name?: string
+  email?: string
+}
+
+export class Repo {
+  SSH_KEY_RELATIVE_PATH = '.ssh/id_rsa'
+
+  constructor(
+    public pathToRepo: string = process.cwd(),
+    public pathToContent: string = ''
+  ) {}
+
+  get contentAbsolutePath() {
+    return path.join(this.pathToRepo, this.pathToContent)
+  }
+
+  get tmpDir() {
+    return path.join(this.contentAbsolutePath, '/tmp/')
+  }
+
+  get sshKeyPath() {
+    return path.join(this.pathToRepo, this.SSH_KEY_RELATIVE_PATH)
+  }
+
+  fileAbsolutePath(fileRelativePath: string) {
+    return path.join(this.contentAbsolutePath, fileRelativePath)
+  }
+
+  fileRelativePath(filepath: string) {
+    return path.posix.join(this.pathToContent, filepath)
+  }
+
+  fileIsInRepo(filepath: string) {
+    const fileIsInRepo = checkFilePathIsInParent(
+      filepath,
+      this.contentAbsolutePath
+    )
+    if (fileIsInRepo) {
+      return true
+    } else {
+      throw new Error(
+        `Failed to write to: ${filepath} \nCannot write outside of the content directory.`
```

Comment:
This method shouldn't throw the error. That should happen in `writeFile` and `deleteFiles`. Right now `deleteFiles` would throw an error saying "Failed to write"

e.g.

```ts
if (this.fileIsInRepo(fileAbsolutePath)) {
  writeFile(fileAbsolutePath, contents)
} else {
  throw new Error(
    `Failed to write to: ${filepath} \nCannot write outside of the content directory.`
  )
}